RecoEgamma/PhotonIdentification/plugins/PhotonMVAEstimatorRunIIFall17.h

```diff
@@ -0,0 +1,127 @@
+    float varE2x2overE5x5;
+    float varSCEta;
+    float varESEnOverRawE; // for endcap MVA only
+    float varESEffSigmaRR; // for endcap MVA only
+    // Pile-up
+    float varRho;
+    // Isolations
+    float varPhoIsoRaw;
+    float varChIsoRaw;
+    float varWorstChRaw;
+
+  };
+  
+  // Constructor and destructor
+  PhotonMVAEstimatorRunIIFall17(const edm::ParameterSet& conf);
+  ~PhotonMVAEstimatorRunIIFall17();
+
+  // Calculation of the MVA value
+  float mvaValue( const edm::Ptr<reco::Candidate>& particle, const edm::Event&) const;
+ 
+  // Utility functions
+  std::unique_ptr<const GBRForest> createSingleReader(const int iCategory, const edm::FileInPath &weightFile);
+  
+  virtual int getNCategories() const { return nCategories; }
+  bool isEndcapCategory( int category ) const;
+  virtual const std::string& getName() const override final { return name_; }
+  virtual const std::string& getTag() const override final { return tag_; }
+  
+  // Functions that should work on both pat and reco electrons
+  // (use the fact that pat::Electron inherits from reco::GsfElectron)
+  std::vector<float> fillMVAVariables(const edm::Ptr<reco::Candidate>& particle, const edm::Event& iEvent) const override;
+  int findCategory( const edm::Ptr<reco::Candidate>& particle ) const override;
+  // The function below ensures that the variables passed to MVA are 
+  // within reasonable bounds
+  void constrainMVAVariables(AllVariables&) const;
+
+  // Call this function once after the constructor to declare
+  // the needed event content pieces to the framework
+  void setConsumes(edm::ConsumesCollector&&) const override;
+  // Call this function once per event to retrieve all needed
+  // event content pices
+  //void getEventContent(const edm::Event& iEvent) override;
+
+  
+ private:
+
+  // MVA name. This is a unique name for this MVA implementation.
+  // It will be used as part of ValueMap names.
+  // For simplicity, keep it set to the class name.
+  const std::string name_ = "PhotonMVAEstimatorRunIIFall17";
+
+  // MVA tag. This is an additional string variable to distinguish
+  // instances of the estimator of this class configured with different
+  // weight files.
+  std::string tag_;
+
+  // Data members
+  std::vector< std::unique_ptr<const GBRForest> > gbrForests_;
+
+  // All variables needed by this MVA
+  const std::string MethodName_;
+  AllVariables allMVAVars_;
+  
+  // This MVA implementation relies on several ValueMap objects
+  // produced upstream. 
+
+  //
+  // Declare all tokens that will be needed to retrieve misc
+  // data from the event content required by this MVA
+  //
+  const edm::InputTag phoChargedIsolationLabel_; 
+  const edm::InputTag phoPhotonIsolationLabel_; 
+  const edm::InputTag phoWorstChargedIsolationLabel_; 
+  const edm::InputTag rhoLabel_;
+
+  // Other objects needed by the MVA
+  EffectiveAreas effectiveAreas_;
+  std::vector<double> phoIsoPtScalingCoeff_;
+  // double          phoIsoCutoff_;
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant

Reply:
Changes implemented here:
https://github.com/lsoffi/cmssw/blob/CMSSW_9_4_0_pre3_TnP/RecoEgamma/PhotonIdentification/plugins/PhotonMVAEstimatorRunIIFall17.h